polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -124,11 +177,34 @@ unsigned getLocalMemoryRemain(gpu::GPUModuleOp &module,
   return localMemoryRemain;
 }
 
-/// Get the required local memory for \p accTy.
+/// Get the required local memory for \p accTy in bytes.
+std::variant<Value, unsigned>
+getReqdLocalMemory(sycl::AccessorType accTy, const WorkGroupSize &workGroupSize,
+                   OpBuilder builder) {
+  Location loc = builder.getUnknownLoc();
+  unsigned elemSize = accTy.getType().getIntOrFloatBitWidth() / 8;
+  std::variant<Value, unsigned> reqdLocalMemory;
```

Comment:
Instead of using `std::variant<Value, unsigned>`  can you use `WorkGroupSize::ElemTy` here.